### PR TITLE
qa_squad clean up 

### PR DESCRIPTION
**Pull request by yzhang123**
clean up squad, simplify QAmodel.test() for single GPU
Signed-off-by: Yang Zhang <yangzhang@nvidia.com>

### CODE REVIEW

File: nemo/collections/nlp/models/question_answering/qa_model.py [hunk 1]
```diff
@@ -178,51 +172,28 @@ def validation_epoch_end(self, outputs):
...
-
-            if self.test_config.output_nbest_file is not None:
-                with open(self.test_config.output_nbest_file, "w") as writer:
-                    writer.write(json.dumps(all_nbest, indent=4) + "\n")
-            if self.test_config.output_prediction_file is not None:
-                with open(self.test_config.output_prediction_file, "w") as writer:
-                    writer.write(json.dumps(all_predictions, indent=4) + "\n")
+        if self._cfg.test_ds.output_nbest_file is not None:
+            with open(self._cfg.test_ds.output_nbest_file, "w") as writer:
+                writer.write(json.dumps(all_nbest, indent=4) + "\n")
```
Comment (ekmb) [inline nemo/collections/nlp/models/question_answering/qa_model.py:193]: check it's done only on rank 0?

File: nemo/collections/nlp/models/question_answering/qa_model.py [hunk 2]
```diff
@@ -178,51 +172,28 @@ def validation_epoch_end(self, outputs):
         return {'val_loss': avg_loss, 'log': tensorboard_logs}
 
     def test_epoch_end(self, outputs):
-        unique_ids = torch.cat([x['test_tensors']['unique_ids'] for x in outputs])
+        unique_ids = torch.cat([x['test_tensors']['unique_ids'] for x in outputs]).cpu().numpy().tolist()
         logits = torch.cat([x['test_tensors']['logits'] for x in outputs])
+        s, e = logits.split(dim=-1, split_size=1)
+        start_logits = s.squeeze().cpu().numpy().tolist()
```
Comment (ekmb) [inline nemo/collections/nlp/models/question_answering/qa_model.py:178]: consider using https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/parts/utils_funcs.py#L34

**Review by ekmb — CHANGES_REQUESTED** [review]

File: nemo/collections/nlp/models/question_answering/qa_model.py (hunk 1 above)
Comment (yzhang123) [reply nemo/collections/nlp/models/question_answering/qa_model.py:193]: shouldn't test() be run on a single gpu anyway? On multi-gpu the result could  be incorrect

Comment (yzhang123) [reply nemo/collections/nlp/models/question_answering/qa_model.py:193]: shouldn't test() be run on a single GPU anyway?

Comment (yzhang123) [reply nemo/collections/nlp/models/question_answering/qa_model.py:194]: ok added check to script, thanks!

File: nemo/collections/nlp/models/question_answering/qa_model.py (hunk 2 above)
Comment (yzhang123) [reply nemo/collections/nlp/models/question_answering/qa_model.py]: done

**Review by ekmb — APPROVED** [review]

**Pull request merged**